lib/juttle-engine.js

```diff
@@ -1,45 +1,28 @@
-var Base = require('extendable-base');
+'use strict';
 var _ = require('underscore');
 var express = require('express');
 var expressWs = require('express-ws');
-var bodyParser = require('body-parser');
-var jobs = require('./job-handlers');
-var paths = require('./path-handlers');
-var prepares = require('./prepare-handlers');
 var logger = require('log4js').getLogger('juttle-engine');
 
 var read_config = require('juttle/lib/config/read-config');
 var Juttle = require('juttle/lib/runtime').Juttle;
-var JuttledErrors = require('./errors');
+var JuttleExpressRoutes = require('./express-routes');
 
 var API_PREFIX = '/api/v0';
 
-var JuttleEngine = Base.extend({
+class JuttleEngine {
 
-    initialize: function(options) {
-        var config = read_config(options);
+    constructor(options) {
+
+        let config = read_config(options);
         Juttle.adapters.configure(config.adapters);
-        var self = this;
 
-        self._config_path = options.config_path;
         this._app = express();
         expressWs(this._app);
 
         this._app.disable('x-powered-by');
 
-        this._root_directory = options.root_directory;
-        this._max_saved_messages = 1024;
-        this._delayed_job_cleanup = 10000;
-
-        if (_.has(config, 'juttled')) {
-            if (_.has(config.juttled, 'max_saved_messages')) {
-                this._max_saved_messages = config.juttled.max_saved_messages;
-            }
-
-            if (_.has(config.juttled, 'delayed_job_cleanup')) {
-                this._delayed_job_cleanup = config.juttled.delayed_job_cleanup;
-            }
-        }
+        JuttleExpressRoutes.add_routes(this._app, options);
 
         // add cors headers, allow ALL origins
         this._app.use(function (req, res, next) {
```

Comment:
I think the CORS logic needs to be applied to the app before you register the routes, otherwise it won't take effect.

Speaking of which I think we should have a test that verifies we properly get the CORS headers, and perhaps it would make more sense to use https://www.npmjs.com/package/cors instead of rolling our own.